### PR TITLE
fix: require a link element in html

### LINKED ISSUE #42324: Passing the challenge while running a wrong code.
**Affected page**

https://www.freecodecamp.org/learn/responsive-web-design/basic-css/import-a-google-font

**Your code**

```

<style>

<link href="https://fonts.googleapis.com/css?family=Lobster" rel="stylesheet" type="text/css">

.red-text {
  color: red;
  
}
h2 {font-family: Lobster;}
p {
  font-size: 16px;
  font-family: monospace;
}
</style>

<h2 class="red-text">CatPhotoApp</h2>
<main>
<p class="red-text">Click here to view more <a href="#">cat photos</a>.</p>

<a href="#"><img src="https://bit.ly/fcc-relaxing-cat" alt="A cute orange cat lying on its back."></a>

<div>
  <p>Things cats love:</p>
  <ul>
    <li>cat nip</li>
    <li>laser pointers</li>
    <li>lasagna</li>
  </ul>
  <p>Top 3 things cats hate:</p>
  <ol>
    <li>flea treatment</li>
    <li>thunder</li>
    <li>other cats</li>
  </ol>
</div>

<form action="https://freecatphotoapp.com/submit-cat-photo">
  <label><input type="radio" name="indoor-outdoor" checked> Indoor</label>
  <label><input type="radio" name="indoor-outdoor"> Outdoor</label><br>
  <label><input type="checkbox" name="personality" checked> Loving</label>
  <label><input type="checkbox" name="personality"> Lazy</label>
  <label><input type="checkbox" name="personality"> Energetic</label><br>
  <input type="text" placeholder="cat photo URL" required>
  <button type="submit">Submit</button>
</form>
</main>


```

**Expected behavior**

Importing `<link>` must be outside the `<style>` tag not inside it. This code above will pass the test cases even tho the font isn't 
applied and imported inside the `<style>` tag.

**Screenshots**

As you can see the lobster font is not applied to the `h2` tag but it still passes all the tests.
bad

This is the right implementation of importing a `link` tag
good

**System (please complete the following information):**
 - Device: Laptop
 - OS: Windows 10
 - Browser: Chrome
 - Version: 90.0.4430.212 (Official Build) (64-bit)

**Additional context**

### PR DESCRIPTION
**Pull request by SethFalco**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42324

This changes the test to look for all `link` element with the `href` attribute, (`link[href]`) and checks if at least one is from `googleapis`.

Technically, the test can be made more strict, but I kept it to the original logic.

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.md [hunk 1]
```diff
@@ -40,7 +40,11 @@ Import the `Lobster` font to your web page. Then, use an element selector to set
 You should import the `Lobster` font.
 
 ```js
-assert(new RegExp('googleapis', 'gi').test(code));
+assert(
+  $('link[href]')
+    .toArray()
+    .some((element) => new RegExp('googleapis', 'gi').test(element.href))
+);
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.md:47]:
```suggestion
assert($('link[href*="googleapis" i]').length);
```
How about this?  Same basic idea, but a little more concise.

Comment (SethFalco) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.md:47]: Didn't know one could make the selector case-insensitive. Much better and nice to know. \\(^-^)/

Comment (ojeytonwilliams) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.md]: Neither did I until a few days ago!  I discovered it when I was trying to write some tests for the new curriculum - it's pretty useful.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by gikf — APPROVED** [review]
LGTM :+1: 

Fixes the problem, tested against code from the issue.

**Pull request merged**